src/Service/Modules/Reports/PartsReceivedReportModule.cs

```diff
@@ -13,66 +12,81 @@
     
     using Linn.Purchasing.Service.Models;
 
+    using Microsoft.AspNetCore.Builder;
     using Microsoft.AspNetCore.Http;
+    using Microsoft.AspNetCore.Routing;
 
-    public class PartsReceivedReportModule : CarterModule
+    public class PartsReceivedReportModule : ICarterModule
     {
-        private readonly ITqmsJobRefService tqmsJobRefService;
-
-        private readonly IPartsReceivedReportFacadeService reportFacadeService;
-
-        public PartsReceivedReportModule(
-            ITqmsJobRefService tqmsJobRefService,
-            IPartsReceivedReportFacadeService reportFacadeService)
+        public void AddRoutes(IEndpointRouteBuilder app)
         {
-            this.tqmsJobRefService = tqmsJobRefService;
-            this.reportFacadeService = reportFacadeService;
-            this.Get("/purchasing/tqms-jobrefs", this.GetJobRefs);
-            this.Get("/purchasing/reports/parts-received", this.GetReport);
-            this.Get("/purchasing/reports/parts-received/export", this.GetExport);
+            app.MapGet("/purchasing/tqms-jobrefs", this.GetJobRefs);
+            app.MapGet("/purchasing/reports/parts-received", this.GetReport);
+            app.MapGet("/purchasing/reports/parts-received/export", this.GetExport);
         }
 
-        private async Task GetJobRefs(HttpRequest request, HttpResponse response)
+        private async Task GetJobRefs(
+            HttpRequest request,
+            HttpResponse response,
+            ITqmsJobRefService tqmsJobRefService)
         {
-            var result = this.tqmsJobRefService.GetMostRecentJobRefs(50);
+            var result = tqmsJobRefService.GetMostRecentJobRefs(50);
             await response.Negotiate(result);
         }
 
-        private async Task GetReport(HttpRequest req, HttpResponse res)
+        private async Task GetReport(
+            HttpRequest req,
+            HttpResponse res,
+            int? supplier,
+            string fromDate,
+            string toDate,
+            string jobref,
+            string orderBy,
+            bool includeNegativeValues,
```

Comment:
this is way nicer than the rq.Query syntax

Reply:
yeah. i agree

Reply:
What do you guys think of doing this to make bigger resource objects too? I hate that when something is null or js has passed something back as a string not an int etc it just fails completely silently with the bind<resource> way, wondering if this would avoid that. It would look pretty horrible with 20 parameters or something though so maybe not